src/GslCore/AstProcess.fs

```diff
@@ -326,19 +326,31 @@ let private inlinePassedArgs (vb: VariableBindings) (fd, fc: FunctionCall) =
         | _ -> error (InternalError(TypeError)) "No function locals node found in function defintion block." fd.body
     | x -> internalTypeMismatch (Some "function body") "Block" x
 
-
 /// Replace a function call with the contents of a function definition.
 let private inlineFunctionCall (s: FunctionInliningState) (node: AstNode) =
     match node with
     | FunctionCall(fcw) when s.insideDefDepth = 0 -> // only do inlining if we're not inside a def
         let fc = fcw.x
-        match s.defs.TryFind(fc.name) with
-        | Some(fd) ->
-            // inline the args into the function call block
-            // this new block replaces the function call
-            checkArgs fd fc node
-            >>= inlinePassedArgs s.vars
-        | None -> error UnresolvedFunction fc.name node
+
+        match fcw.Pos with
```

Comment:
Hmm; why is this match on the optionalized version of the source position?  Don't we want to take the entire list and concatenate it with that on the node rather than pushing just the head?  Seems like this *might* actually be the source of the missing intermediate call sites?  Changing this to `fcw.positions` (and possibly removing the branch that would fail if the list were empty which currently catches None) will probably get us there!

Reply:
Good call - that fixes the missing line number problem.  I have re-enabled the nested call test and made it more detailed (checks number and position of values).